src/OpenTelemetry.Contrib.Instrumentation.AWS/Implementation/AWSTracingPipelineHandler.cs

```diff
@@ -0,0 +1,270 @@
+        {
+            var activity = this.ProcessBeginRequest(executionContext);
+            try
+            {
+                base.InvokeSync(executionContext);
+            }
+            catch (Exception ex)
+            {
+                if (activity != null)
+                {
+                    this.ProcessException(activity, ex);
+                }
+
+                throw;
+            }
+            finally
+            {
+                if (activity != null)
+                {
+                    this.ProcessEndRequest(executionContext, activity);
+                }
+            }
+        }
+
+        public override async Task<T> InvokeAsync<T>(IExecutionContext executionContext)
+        {
+            T ret = null;
+
+            var activity = this.ProcessBeginRequest(executionContext);
+            try
+            {
+                ret = await base.InvokeAsync<T>(executionContext).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                if (activity != null)
+                {
+                    this.ProcessException(activity, ex);
+                }
+
+                throw;
+            }
+            finally
+            {
+                if (activity != null)
+                {
+                    this.ProcessEndRequest(executionContext, activity);
+                }
+            }
+
+            return ret;
+        }
+
+        private Activity ProcessBeginRequest(IExecutionContext executionContext)
+        {
+            Activity activity = null;
+
+            var requestContext = executionContext.RequestContext;
+            var service = this.GetAWSServiceName(requestContext);
+            var operation = this.GetAWSOperationName(requestContext);
+
+            activity = AWSSDKActivitySource.StartActivity(service + "." + operation, ActivityKind.Client);
+
+            if (activity == null)
+            {
+                return null;
+            }
+
+            if (this.options.SuppressDownstreamInstrumentation)
+            {
+                SuppressInstrumentationScope.Enter();
+            }
+
+            if (activity.IsAllDataRequested)
+            {
+                activity.SetTag(AWSSemanticConventions.AttributeAWSServiceName, service);
+                activity.SetTag(AWSSemanticConventions.AttributeAWSOperationName, operation);
+                var client = executionContext.RequestContext.ClientConfig;
+                if (client != null)
+                {
+                    var region = client.RegionEndpoint?.SystemName;
+                    activity.SetTag(AWSSemanticConventions.AttributeAWSRegion, region ?? AWSSDKUtils.DetermineRegion(client.ServiceURL));
+                }
+
+                this.AddRequestSpecificInformation(activity, requestContext, service);
+            }
+
+            AwsPropagator.Inject(new PropagationContext(activity.Context, Baggage.Current), requestContext.Request.Headers, Setter);
```

Comment:
as mentioned in other comment, I am not convinced that an instrumentation should hardcode to a particular propagation format. 

(Also the AwsPropagator doesn't look like it propagate Baggage. If thats the case you'd need someone else to propagate Baggage)

Reply:
None of the downstream AWS managed services which this request will go to understands Baggage. They would simply look for the [tracing header](https://docs.aws.amazon.com/xray/latest/devguide/xray-concepts.html#xray-concepts-tracingheader) present in the http header of the request. So the propagated Baggage or any other context would not be processed downstream.